system/class/Database/Database.php

```diff
@@ -352,6 +352,10 @@ static function val($value): string
             return $value->getSql();
         }
 
+        if (is_bool($value)) {
+            return (int) $value;
```

Comment:
My bad, there is a `string` return type.

```suggestion
            return $value ? '1' : '0';
```